docs/networking/acns-lab.md

```diff
@@ -198,79 +288,811 @@ Let's start by applying a new network policy to cause some chaos in the network.
+
+---
+
+**Summary of Test Scenarios:**
+
+| Scenario | Test Type | Protocol/Layer | Result |
+|----------|-----------|----------------|--------|
+| 1 | External → store-front | L3/L4 (TCP/IP) | ❌ Ingress blocked by chaos policy |
+| 2 | microsoft.com | L7 (HTTPS/FQDN) | ✅ Network OK (HTTP 403 from server) |
+| 3 | api.github.com | L7 (DNS/FQDN) | ❌ DNS query blocked (missing from DNS patterns) |
+| 4 | GET /api/products | L7 (HTTP method) | ✅ Allowed by L7 policy |
+| 4b | POST /api/products | L7 (HTTP method) | ❌ Blocked by L7 policy (only GET allowed) |
+| 5 | store-front → order-service | L3/L4 (TCP internal) | ✅ Internal communication works |
+| 6 | nslookup order-service | DNS (internal) | ✅ Internal DNS resolution works |
+
+<div class="info" data-title="Key Insight">
+
+> **FQDN Policy Requirements**: Notice that `api.github.com` is in the `toFQDNs` list but still fails. This demonstrates a critical Cilium requirement: you need **BOTH** DNS rules (to allow the DNS query) **AND** toFQDNs rules (to allow the connection). Having only toFQDNs without DNS rules means the DNS query gets blocked first, and the connection is never attempted. This is a common misconfiguration that flow logs will help you identify.
+
+</div>
+
+**What You Just Simulated:**
+
+1. **External Access Failures**: The chaos policy is blocking all ingress traffic to store-front
+2. **Selective FQDN Filtering**: The FQDN policy allows `*.microsoft.com`, but blocks `api.github.com` at the DNS level
+3. **L7 HTTP Method Filtering**: The L7 policy allows GET requests but blocks POST requests to the product-service API, demonstrating application-layer control
+4. **Internal Service Mesh**: Pod-to-pod communication within the namespace continues to work normally
+
+> **Key Insight - FQDN Policy Requirements:** Notice that `api.github.com` is in the `toFQDNs` list but still fails. This demonstrates a critical Cilium requirement: you need **BOTH** DNS rules (to allow the DNS query) **AND** toFQDNs rules (to allow the connection). Having only toFQDNs without DNS rules means the DNS query gets blocked first, and the connection is never attempted. This is a common misconfiguration that flow logs will help you identify.
+
+<div class="info" data-title="Important">
+
+> **Understanding FQDN Policy Requirements**: For FQDN filtering to work in Cilium, you need **both** components:
+>
+> 1. **DNS rules** (port 53) - Allow the DNS query to resolve the domain name
+> 2. **toFQDNs rules** - Allow the connection to the resolved IP address
+>
+> In this policy, `*.microsoft.com` works because it's in **both** the DNS patterns and toFQDNs rules. However, domains like `api.github.com`, `bing.com`, and `google.com` fail because they're missing from the DNS `matchPattern` rules - the DNS query gets blocked first, so the pod never gets to attempt the actual connection. This is a common misconfiguration that container network flow logs help you identify quickly.
+
+</div>
```

Comment:
```suggestion
:::danger Important

**Understanding FQDN Policy Requirements**: For FQDN filtering to work in Cilium, you need **both** components:

1. **DNS rules** (port 53) - Allow the DNS query to resolve the domain name
2. **toFQDNs rules** - Allow the connection to the resolved IP address

In this policy, `*.microsoft.com` works because it's in **both** the DNS patterns and toFQDNs rules. However, domains like `api.github.com`, `bing.com`, and `google.com` fail because they're missing from the DNS `matchPattern` rules - the DNS query gets blocked first, so the pod never gets to attempt the actual connection. This is a common misconfiguration that container network flow logs help you identify quickly.

:::
```